asset_dashboard/views.py

```diff
@@ -41,22 +42,41 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class AddEditProjectFormView(FormView):
-    template_name = 'asset_dashboard/partials/add_edit_project_form.html'
+class AddProjectFormView(CreateView):
+    template_name = 'asset_dashboard/partials/add_project_modal_form.html'
     form_class = ProjectForm
 
     def post(self, request, *args, **kwargs):
         form = self.form_class(request.POST)
 
         if form.is_valid():
-            self.fcc_form = form.save()
-           
-            # redirect to the new Project's detail page
-            return HttpResponseRedirect(reverse('project-detail', kwargs={'pk': self.fcc_form.id}))
+            project = Project.objects.create(**form.cleaned_data)
+            return HttpResponseRedirect(reverse('project-detail', kwargs={'pk': project.pk}))
         else:
-            return HttpResponseBadRequest('Form data was invalid.')
+            return HttpResponseBadRequest('Form is invalid.')
 
 
-class ProjectDetailView(DetailView):
+class ProjectDetailView(DetailView, edit.FormMixin):
```

Comment:
i would think that you might want this to be an updateview. is there a reason you didn't go down that road. 

it would seem to avoid having to manually mange the get_context_data stuff?

Reply:
I've refactored to use the UpdateView. It's much better, thank you. It also fixed the overwrite bug. Here is the commit: d77f6f5.